src/install/package.ts

```diff
@@ -207,6 +207,30 @@ export async function parseTarget(
   if (versionIndex !== -1) alias = pkg.pkgName.slice(0, versionIndex);
   else alias = pkg.pkgName;
 
+  // If no version is specified, we fallback to the constraints in the parent
+  // package config if they exist:
+  const pcfg = await resolver.getPackageConfig(parentPkgUrl.href);
```

Comment:
Main change, if there's no version specified in a `generator.install(<spec>, ...)` call, then this looks up version constraints from the base `package.json` (if it exists). I think this is actually the root cause of a few bugs, including #211.